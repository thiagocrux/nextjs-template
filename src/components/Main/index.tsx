import * as Styled from './styles';

export type Props = {
  title?: string;
  description?: string;
};

export default function Main({
  title = 'Next.js Template',
  description = 'Start your new project from here...',
}: Props) {
  return (
    <Styled.Wrapper data-testid="main">
      <Styled.Title>{title}</Styled.Title>
      <Styled.Description>{description}</Styled.Description>
    </Styled.Wrapper>
  );
}
