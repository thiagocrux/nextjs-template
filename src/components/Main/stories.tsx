import { Meta, StoryObj } from '@storybook/react';

import Main from '.';

export default {
  title: 'Main',
  component: Main,
  argTypes: {
    title: {
      control: { type: 'text' },
    },
    description: {
      control: { type: 'text' },
    },
  },
} as Meta;

export const Default: StoryObj = {};
