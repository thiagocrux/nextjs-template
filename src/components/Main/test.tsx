import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render the heading with the right text and backgroudn color', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /hello, world!/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toHaveStyle({
      'background-color': '#ffffff',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
