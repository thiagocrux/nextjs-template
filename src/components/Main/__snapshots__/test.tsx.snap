// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading with the right text and backgroudn color 1`] = `
.c0 {
  color: red;
  background-color: #ffffff;
}

<main
  class="c0"
  data-testid="main"
>
  <h1>
    Hello, world!
  </h1>
</main>
`;
